CRM/Member/Form/MembershipRenewal.php

```diff
@@ -339,6 +339,12 @@ public function buildQuickForm() {
 
     $this->applyFilter('__ALL__', 'trim');
 
+    // CRM-20571
+    // Get the Join Date from Membership info as it is not available in the Renew form
+    $joinDate = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_Membership', $this->_id, 'join_date');
+    // Save its value in a hidden field
+    $this->addElement('hidden', 'join_date', $joinDate);
```

Comment:
@amsharma9 Instead of creating a hidden field and storing values, can we not check the `join_date` in `formRule()` itself using membership id? 

You can pass `$this` to [addFormRule()](https://github.com/civicrm/civicrm-core/blob/master/CRM/Member/Form/MembershipRenewal.php#L407) in order to get membership id. Something like -

    $this->addFormRule(array('CRM_Member_Form_MembershipRenewal', 'formRule'), $this);

and then add params to `formRule()` function

    public static function formRule($params, $files, $self) {

You should get membership id in `$self->_id`.